.github/workflows/validate.yaml

```diff
@@ -0,0 +1,46 @@
+name: validate-files
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
```

Comment:
Why narrow this job down to these types?

Reply:
The idea was to have the CI auto-run for every PR to the repo and this would prevent the CI to run on PRs that weren't ready. I know we discarded the idea of having it auto-run on every third-party PR, but I think this is still good, wdyt?

Reply:
What's wrong with having it run on every PR?

Reply:
The initial plan for this was that if we would change the permissions so that we wouldn't need to approve a workflow for it to run, so by doing these we would make sure only these types of PR would auto-run the workflow

Reply:
We need to approve the workflows for security reasons, I don't think narrowing down the PR type will help. I recommend cutting this but it's up to you.

Reply:
Removed PR types!